### DISCUSSION TITLE
Add User Confirmation for Start/Stop/Uninstall/etc. Commands?

### DISCUSSION THREAD

**Discussion by jgriegershs** [Siemens-Healthineers/K2s#227] [Ideas]
As users may have workloads running on the *K2s* cluster that shall not be interrupted, does it make sense to add user confirmation to CLI commands that potentially interrupt workloads or delete data at rest/in transit?

Commands with interruption potential are `k2s start`, `k2s stop`, `k2s uninstall`, etc.

**Comment by krotz-dieter** [comment]
From my point of view:
- uninstall should have an confirmation option with possibility of omitting it. 
- start/stop does not do any harm, it keeps workloads: it shall not have any confirmation
- addons disable: also same as uninstall
- system update: also same as uninstall
All others keep without confirmation.

**Reply by mahenarayan** [reply]
@jgriegershs  
Yes, I agree with Dieter's points, and I have some more commands which would require user confirmation. 
Complete summary including Dieter's list:

```
k2s uninstall
k2s addons disable
k2s system upgrade 
k2s system reset 
k2s image clean

```

Default behavior is to ask for user confirmation and proceed automatically when flag `-f ` or `--force` is passed.
Please create an issue out of this discussion.

**Reply by jgriegershs** [reply]
I disagree with `k2s stop` doing no harm, since it interrupts workloads, makes hosted services unavailable and potentially kills data in transit.

If the cluster is already started, `k2s start` will restart it, causing the same aforementioned problems. One mitigation though would be to remove the implicit cluster restart from this command.
